Sources/Testing/Parameterization/TypeInfo.swift

```diff
@@ -99,6 +99,48 @@ extension TypeInfo {
   /// An in-memory cache of fully-qualified type name components.
   private static let _fullyQualifiedNameComponentsCache = Locked<[ObjectIdentifier: [String]]>()
 
+  /// Split the given fully-qualified type name into its components.
+  ///
+  /// - Parameters:
+  ///   - fullyQualifiedName: The string to split.
+  ///
+  /// - Returns: The components of `fullyQualifiedName` as substrings thereof.
+  static func fullyQualifiedNameComponents(ofTypeWithName fullyQualifiedName: String) -> [String] {
+    var components = [Substring]()
+
+    var inRawIdentifier = false
+    var componentStartIndex = fullyQualifiedName.startIndex
+    for i in fullyQualifiedName.indices {
+      let c = fullyQualifiedName[i]
+      if c == "`" {
+        inRawIdentifier.toggle()
+      } else if c == "." && !inRawIdentifier {
+        components.append(fullyQualifiedName[componentStartIndex ..< i])
+        componentStartIndex = fullyQualifiedName.index(after: i)
+      }
+    }
+    components.append(fullyQualifiedName[componentStartIndex...])
+
+    // If a type is extended in another module and then referenced by name,
+    // its name according to the String(reflecting:) API will be prefixed with
+    // "(extension in MODULE_NAME):". For our purposes, we never want to
+    // preserve that prefix.
+    if let firstComponent = components.first, firstComponent.starts(with: "(extension in "),
+       let moduleName = firstComponent.split(separator: ":", maxSplits: 1).last {
```

Comment:
This will presumably mis-parse:

```
(extension in `Foo:Bar`):Quux
```

Which means we _do_ need to walk this substring looking for the first `:` _outside_ any pair of backticks.

Reply:
Right. And in my use of raw identifiers with Bazel, *every* module name will have a colon in it 😁 

I wonder if a simple regex would be the best approach here, to check for a prefix of ``\(extension in (`[^`]+`|[^)]+)\)``.

Reply:
Ah, I see that you did add this, GitHub just loaded an outdated diff when I clicked the link from the e-mail...